storage/series/reader.go

```diff
@@ -133,17 +137,11 @@ func (r Reader) readersWithBlocksMapAndBuffer(
 		case r.retriever != nil:
 			// Try to stream from disk
 			if r.retriever.IsBlockRetrievable(blockAt) {
-				if clonedID == nil {
-					// Clone ID as the block retriever uses the ID async so we cannot
-					// be sure about owner not finalizing the cloneableID passed to
-					// the Reader
-					clonedID = r.opts.IdentifierPool().Clone(r.cloneableID)
-					ctx.RegisterFinalizer(clonedID)
-				}
-				// TODO(r): Specify a non-nil OnRetrieveBlock callback to cache the
-				// block back to a series in the shard on a read.
-				var onRetrieve block.OnRetrieveBlock
-				stream, err := r.retriever.Stream(clonedID, blockAt, onRetrieve)
+				// Clone ID as the block retriever uses the ID async so we cannot
+				// be sure about owner not finalizing the cloneableID passed to
+				// the Reader
+				clonedID := r.opts.IdentifierPool().Clone(r.cloneableID)
+				stream, err := r.retriever.Stream(clonedID, blockAt, r.onRetrieve)
```

Comment:
What is the lifecycle management for the clonedID now that its not finalized by the ctx?

Reply:
Yeah so basically the callee (OnRetrieveBlock) now needs to finalize it, the issue with cloning it and pinning it to the lifecycle of the context is that it might not be safe to release the ID once the request finishes because the OnRetrieveBlock callee needs to use it asynchronously.

Reply:
You know what, I might move this solely into the retriever - let me attempt this.